scvi/model/base/_demixin.py

```diff
@@ -0,0 +1,473 @@
+        batch_correction: bool = False,
+        batchid1: Optional[Iterable[str]] = None,
+        batchid2: Optional[Iterable[str]] = None,
+        fdr_target: float = 0.05,
+        silent: bool = False,
+        pseudocounts: float = 0.0,
+        fn_kwargs: Optional[dict] = None,
+        importance_sampling: Optional[bool] = False,
+        **kwargs,
+    ) -> pd.DataFrame:
+        r"""
+        A unified method for differential expression analysis.
+
+        Implements `"vanilla"` DE [Lopez18]_ and `"change"` mode DE [Boyeau19]_.
+        When using the change method, uses either the plugin estimator
+        or importance sampling for improved FDR control.
+
+        Parameters
+        ----------
+        {doc_differential_expression}
+        **kwargs
+            Keyword args for :func:`scvi.utils.DifferentialComputation.get_bayes_factors`
+
+        Returns
+        -------
+        Differential expression DataFrame.
+        """
+        adata = self._validate_anndata(adata)
+        adata.uns["_scvi"]["_requires_validation"] = False
+        fn_kwargs = dict() if fn_kwargs is None else fn_kwargs
+        col_names = _get_var_names_from_setup_anndata(adata)
+        if importance_sampling:
+            model_fn = partial(
+                self.get_population_expression,
+                return_numpy=True,
+                batch_size=batch_size,
+                **fn_kwargs,
+            )
+        else:
+            model_fn = partial(
+                self.get_normalized_expression,
+                return_numpy=True,
+                n_samples=1,
+                batch_size=batch_size,
+            )
+
+        result = _de_core(
+            adata,
+            model_fn,
+            groupby,
+            group1,
+            group2,
+            idx1,
+            idx2,
+            all_stats,
+            scrna_raw_counts_properties,
+            col_names,
+            mode,
+            batchid1,
+            batchid2,
+            delta,
+            batch_correction,
+            fdr_target,
+            silent,
+            pseudocounts=pseudocounts,
+            use_permutation=True,
+            **kwargs,
+        )
+
+        return result
+
+    @torch.no_grad()
+    def get_population_expression(
+        self,
+        adata: Optional[AnnData] = None,
+        indices: Optional[Sequence[int]] = None,
+        n_samples: int = 25,
+        n_samples_overall: int = None,
+        batch_size: Optional[int] = 64,
+        filter_cells: bool = True,
+        transform_batch: Optional[Sequence[Union[Number, str]]] = None,
+        return_numpy: Optional[bool] = False,
+        marginal_n_samples_per_pass: int = 500,
+        n_mc_samples_px: int = 5000,
+        n_cells_per_chunk: Optional[int] = 500,
+        max_chunks: Optional[int] = None,
+        normalized_expression_key: str = "px_scale",
+    ) -> np.ndarray:
+        """
+        Computes importance-weighted expression levels within a subpopulation.
```

Comment:
you are sampling from an importance weighted version of the posterior predictive right? It should be better explained below how else this can be used, and why it's different than `get_normalized_expression` so people aren't confused.